docs/modules/ROOT/pages/variables-and-params.adoc

```diff
@@ -0,0 +1,384 @@
+const movie = new Cypher.Node({ labels: ["Movie"] });
+const titleProp = movie.property(movie);
+const titleParam1 = new Cypher.Param("The Matrix");
+const titleParam2 = new Cypher.Param("The Matrix")
+
+const query = new Cypher.Match(movie).where(Cypher.eq(titleProp, titleParam1)).return(titleParam2);
+
+const {cypher, params} = query.build();
+```
+
+_Cypher_
+```cypher
+MATCH (this0:Movie)
+WHERE this0[this0] = $param0
+RETURN this0, $param0
+```
+
+_Params_
+```js
+{
+    param0: "The Matrix"
+}
+```
+
+
+== Literal
+Literal values can be defined with `Cypher.Literal`. Literals behave like Parameters, but it will inject the value provided directly into the Cypher, serializing it as needed.
+
+For instance:
+
+```javascript
+const movie = new Cypher.Node({ labels: ["Movie"] });
+const titleProp = movie.property(movie);
+const titleLiteral = new Cypher.Literal("The Matrix")
+
+const query = new Cypher.Match(movie).where(Cypher.eq(titleProp, titleParam)).return(titleParam);
+
+const {cypher, params} = query.build();
+```
+
+_Cypher_
+```cypher
+MATCH (this0:Movie)
+WHERE this0[this0] = "The Matrix"
+RETURN this0, "The Matrix"
+```
+
+_Params_
+```js
+{ }
+```
+
+Note how the value `The Matrix` is not injected directly, but correctly serialized to a string in Cypher. The following values are supported by `Literal`:
+
+* String: `Cypher.Literal("Hello")` -> `"Hello"`
+* Number: `Cypher.Literal(5)` -> `5`
+* Boolean: `Cypher.Literal(true)` -> `true`
+* Array: `Cypher.Literal([5, "Hello"])` -> `[5, "Hello"]`
+* Null: `Cypher.Literal(null)` -> `NULL`
+
+
+[NOTE]
+====
+Note that Literal does not escape the values and code injection is a risk. It is generally recommended to use `Cypher.Param` instead.
+====
+
+=== NULL
+
+As a shortcut for `new Cypher.Literal(null)`, the constant `Cypher.Null` is available. This will be translated to `NULL`.
+
+
+== Aliasing
+// Maybe aliasing can be moved to a how-to guide
+Variables are commonly used for aliasing in a `WITH` or `RETURN` statement. To do that, simply pass a tuple of the value and its alias.
+
+=== Aliasing to a string
+
+```javascript
+const node = new Cypher.Node({
+    labels: ["Movie"],
+});
+const withQuery = new Cypher.With([node, "my-alias"]);
+```
+
+```cypher
+WITH this0 AS my-alias
+```
+
+=== Aliasing to a variable
+
+Instead of an exact string, you can alias to a `Cypher.Variable` so it can be reused as any other variable:
+
+```javascript
+const movieNode = new Cypher.Node({ labels: ["Movie"] });
+const myVar = new Cypher.Variable();
+const match = new Cypher.Match(movieNode).with([movieNode, myVar]).return([myVar, "Film"]);
+```
+
+```
+MATCH (this0:`Movie`)
+WITH this0 AS var1
+RETURN var1 AS Film 
+```
+
+In the previous example, after a `MATCH` the node variable `this0` is aliased to a variable with an arbitrary name (`var1`) in a `WITH` statement. Finally, in the `RETURN` the variable is aliased to the specific name `Film` that will be returned.
+
+// TODO
+// == Environment
```

Comment:
Also wondering if this should be in the variables and params page or be separated. Since I don't know what's this is about, I just leave it as a note for you to judge. :)

Reply:
mmm, it is closely related, it could be on its own page, but I don't expect this section to be too big

When we decide to tackle this section (it is not really a priority, that's why I left it) we can decide on where to put it